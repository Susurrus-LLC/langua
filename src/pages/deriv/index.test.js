import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { PureDeriv as Deriv } from '.'

describe('Deriv', () => {
  const data = {
    site: {
      siteMetadata: {
        toolInfo: {
          deriv: {
            title: 'Deriv',
            link: 'deriv',
            description: 'Deriv description'
          }
        }
      }
    }
  }

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Deriv data={data} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders accurately', () => {
    const tree = renderer.create(<Deriv data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
