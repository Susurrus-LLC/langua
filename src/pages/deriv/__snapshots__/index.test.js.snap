// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deriv renders accurately 1`] = `
<div
  className="app"
>
  <header
    className="header"
  >
    <div
      className="titleArea"
    >
      <h1
        className="siteTitle"
      >
        <a
          className="titleLink"
          href="/"
        />
      </h1>
      <p
        className="subtitle"
      />
    </div>
    <nav
      className="nav"
    >
      <ul
        className="menu"
      >
        <li
          className="navLink"
        >
          <a
            href="/gen"
          >
            Gen
          </a>
        </li>
        <li
          className="navLink disabled"
        >
          <a
            href="/morph"
          >
            Morph
          </a>
        </li>
        <li
          className="navLink"
        >
          <a
            href="/deriv"
          >
            Deriv
          </a>
        </li>
        <li
          className="navLink"
        >
          <a
            href="/frequen"
          >
            Frequen
          </a>
        </li>
        <li
          className="navLink disabled"
        >
          <a
            href="/tree"
          >
            Tree
          </a>
        </li>
        <li
          className="navLink disabled"
        >
          <a
            href="/lex"
          >
            Lex
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    className="main"
  >
    <a
      className="btn buttonLink"
      href="/deriv/help"
    >
      Help
    </a>
    <h2
      className="toolTitle"
    >
      undefinedDeriv
    </h2>
    <form
      className="form"
      name="deriv-form"
    >
      <div
        className="lexemes part"
      >
        <h5
          className="sectionTitle"
        >
          Lexemes
        </h5>
        <div
          className="derivRow"
        >
          <h6
            className="columnLabel wordLabel"
          >
            Word
          </h6>
          <h6
            className="columnLabel definitionLabel"
          >
            Gloss
          </h6>
        </div>
        <div
          className="derivRow"
        >
          <input
            aria-label="Word for lexeme 0"
            className="lexemeInput"
            id="lw0"
            name="lw0"
            onChange={[Function]}
            type="text"
            value="nzi"
          />
          <input
            aria-label="Gloss for lexeme 0"
            className="lexemeDefInput"
            id="ld0"
            name="ld0"
            onChange={[Function]}
            type="text"
            value="tree"
          />
          <div
            className="clearButton"
          >
            <button
              className="btn danger small"
              id="lc0"
              onClick={[Function]}
              role="button"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          className="derivRow"
        >
          <input
            aria-label="Word for lexeme 1"
            className="lexemeInput"
            id="lw1"
            name="lw1"
            onChange={[Function]}
            type="text"
            value="ilu"
          />
          <input
            aria-label="Gloss for lexeme 1"
            className="lexemeDefInput"
            id="ld1"
            name="ld1"
            onChange={[Function]}
            type="text"
            value="flower"
          />
          <div
            className="clearButton"
          >
            <button
              className="btn danger small"
              id="lc1"
              onClick={[Function]}
              role="button"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          className="derivRow"
        >
          <input
            aria-label="Word for lexeme 2"
            className="lexemeInput"
            id="lw2"
            name="lw2"
            onChange={[Function]}
            type="text"
            value="ru"
          />
          <input
            aria-label="Gloss for lexeme 2"
            className="lexemeDefInput"
            id="ld2"
            name="ld2"
            onChange={[Function]}
            type="text"
            value="person"
          />
          <div
            className="clearButton"
          >
            <button
              className="btn danger small"
              id="lc2"
              onClick={[Function]}
              role="button"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          className="derivRow"
        >
          <input
            aria-label="Word for lexeme 3"
            className="lexemeInput"
            id="lw3"
            name="lw3"
            onChange={[Function]}
            type="text"
            value="kwərə"
          />
          <input
            aria-label="Gloss for lexeme 3"
            className="lexemeDefInput"
            id="ld3"
            name="ld3"
            onChange={[Function]}
            type="text"
            value="goat"
          />
          <div
            className="clearButton"
          >
            <button
              className="btn danger small"
              id="lc3"
              onClick={[Function]}
              role="button"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          className="derivRow"
        >
          <input
            aria-label="Word for lexeme 4"
            className="lexemeInput"
            id="lw4"
            name="lw4"
            onChange={[Function]}
            type="text"
            value="lɛtsi"
          />
          <input
            aria-label="Gloss for lexeme 4"
            className="lexemeDefInput"
            id="ld4"
            name="ld4"
            onChange={[Function]}
            type="text"
            value="sword"
          />
          <div
            className="clearButton"
          >
            <button
              className="btn danger small"
              id="lc4"
              onClick={[Function]}
              role="button"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          className="addingRow"
        >
          <button
            className="btn success small"
            id="lexAdd"
            onClick={[Function]}
            role="button"
            type="button"
          >
            Add
          </button>
        </div>
      </div>
      <div
        className="derivations part"
      >
        <h5
          className="sectionTitle"
        >
          Derivational Affixes
        </h5>
        <div
          className="derivRow"
        >
          <h6
            className="columnLabel affixLabel"
          >
            Affix
          </h6>
          <h6
            className="columnLabel glossLabel"
          >
            Gloss
          </h6>
          <h6
            className="columnLabel definitionLabel"
          >
            Definition
          </h6>
        </div>
        <div
          className="derivRow"
        >
          <input
            aria-label="Affix 0"
            className="derivationInput"
            id="da0"
            name="da0"
            onChange={[Function]}
            type="text"
            value="-vo"
          />
          <input
            aria-label="Gloss for affix 0"
            className="derivationGlossInput"
            id="dl0"
            name="dl0"
            onChange={[Function]}
            type="text"
            value="person"
          />
          <input
            aria-label="Definition for affix 0"
            className="derivationDefInput"
            id="dd0"
            name="dd0"
            onChange={[Function]}
            type="text"
            value="Creates a noun referring to a person associated with the base."
          />
          <div
            className="clearButton"
          >
            <button
              className="btn danger small"
              id="dc0"
              onClick={[Function]}
              role="button"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          className="derivRow"
        >
          <input
            aria-label="Affix 1"
            className="derivationInput"
            id="da1"
            name="da1"
            onChange={[Function]}
            type="text"
            value="-lu"
          />
          <input
            aria-label="Gloss for affix 1"
            className="derivationGlossInput"
            id="dl1"
            name="dl1"
            onChange={[Function]}
            type="text"
            value="place"
          />
          <input
            aria-label="Definition for affix 1"
            className="derivationDefInput"
            id="dd1"
            name="dd1"
            onChange={[Function]}
            type="text"
            value="Creates a noun denoting the location of the base."
          />
          <div
            className="clearButton"
          >
            <button
              className="btn danger small"
              id="dc1"
              onClick={[Function]}
              role="button"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          className="derivRow"
        >
          <input
            aria-label="Affix 2"
            className="derivationInput"
            id="da2"
            name="da2"
            onChange={[Function]}
            type="text"
            value="ña-"
          />
          <input
            aria-label="Gloss for affix 2"
            className="derivationGlossInput"
            id="dl2"
            name="dl2"
            onChange={[Function]}
            type="text"
            value="abstract"
          />
          <input
            aria-label="Definition for affix 2"
            className="derivationDefInput"
            id="dd2"
            name="dd2"
            onChange={[Function]}
            type="text"
            value="Creates an abstract noun."
          />
          <div
            className="clearButton"
          >
            <button
              className="btn danger small"
              id="dc2"
              onClick={[Function]}
              role="button"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          className="derivRow"
        >
          <input
            aria-label="Affix 3"
            className="derivationInput"
            id="da3"
            name="da3"
            onChange={[Function]}
            type="text"
            value="k-sa"
          />
          <input
            aria-label="Gloss for affix 3"
            className="derivationGlossInput"
            id="dl3"
            name="dl3"
            onChange={[Function]}
            type="text"
            value="collective"
          />
          <input
            aria-label="Definition for affix 3"
            className="derivationDefInput"
            id="dd3"
            name="dd3"
            onChange={[Function]}
            type="text"
            value="Creates a noun referring to a collection of bases."
          />
          <div
            className="clearButton"
          >
            <button
              className="btn danger small"
              id="dc3"
              onClick={[Function]}
              role="button"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          className="addingRow"
        >
          <button
            className="btn success small"
            id="derAdd"
            onClick={[Function]}
            role="button"
            type="button"
          >
            Add
          </button>
        </div>
      </div>
      <div
        className="control part"
      >
        <div
          className="controlSide controlLeft"
        >
          <div
            className="controlPiece"
          >
            <button
              className="btn neutral"
              onClick={[Function]}
              role="button"
              type="submit"
            >
              Derive
            </button>
          </div>
          <div
            className="controlPiece"
          >
            <label
              htmlFor="words"
            >
              number of words to derive:
            </label>
            <input
              className="wordsInput"
              id="words"
              max="9999"
              min="1"
              name="words"
              onChange={[Function]}
              type="number"
              value={5}
            />
          </div>
          <div
            className="controlPiece"
          >
            <label
              htmlFor="type"
            >
              output type:
            </label>
            <select
              id="type"
              name="type"
              onChange={[Function]}
              value="any"
            >
              <option
                value="any"
              >
                any base, any derivation
              </option>
              <option
                value="single base"
              >
                single base
              </option>
              <option
                value="single derivation"
              >
                single derivation
              </option>
            </select>
          </div>
        </div>
        <div
          className="controlSide controlRight"
        >
          <div
            className="controlPiece"
          >
            <button
              className="btn success"
              id="save"
              onClick={[Function]}
              role="button"
              type="button"
            >
              Save
            </button>
          </div>
          <div
            className="controlPiece"
          >
            <button
              className="btn"
              id="open"
              onClick={[Function]}
              role="button"
              type="button"
            >
              Open
            </button>
            <input
              className="hidden"
              id="file"
              name="file"
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </div>
    </form>
    <div
      className="results"
    >
      <div
        className="output"
      />
      <div
        className="stats"
      >
        <p
          className="statsText"
        >
          words: 0; maximum derivations possible: 0
        </p>
      </div>
    </div>
    <div
      className="Toastify"
    />
  </main>
  <footer
    className="footer"
  >
    <p
      className="copyright"
    >
      <a
        href="undefined/blob/master/CHANGELOG.md"
        rel="noopener noreferrer"
        target="_blank"
      >
        Version 
        <i
          className="fas fa-code"
        />
         
      </a>
      . Built by
       
      <a
        rel="noopener noreferrer"
        target="_blank"
      />
      , copyright © 
      2017–2020
       under the
       
      <a
        href="https://choosealicense.com/licenses/agpl-3.0/"
        rel="noopener noreferrer"
        target="_blank"
      >
        AGPL-3.0 
        <i
          className="fas fa-gavel"
        />
         license
      </a>
      .
       
      <a
        rel="noopener noreferrer"
        target="_blank"
      >
        View this project on 
        <i
          className="fab fa-github"
        />
         GitHub.
      </a>
    </p>
  </footer>
</div>
`;
